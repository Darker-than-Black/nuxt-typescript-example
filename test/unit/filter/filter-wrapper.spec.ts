import VueTestInstance from '~/test/factory/vue-instance'
import FilterWrapper from '~/components/filter/filter-wrapper.vue'

describe('filter-wrapper.vue', () => {
  const nameSlot: string = '<h1>Header</h1>'
  const slotContent: string = '<p>lorem ipsum</p>'
  let wrapper: any

  beforeEach(() => {
    wrapper = VueTestInstance(FilterWrapper, {
      isShallow: false,
      directives: {
        'click-outside': () => {}
      },
      options: {
        slots: { name: nameSlot, default: slotContent },
        props: {
          label: 'Filter Wrapper',
          showOptions: true,
          isActive: false,
          hideOptions: jest.fn()
        }
      }
    })
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('check emit "toggleOptionsBar"', () => {
    wrapper.find('p.filter__label').trigger('click')
    expect(wrapper.emitted()['toggle-options-bar']).toBeTruthy()
  })
})
