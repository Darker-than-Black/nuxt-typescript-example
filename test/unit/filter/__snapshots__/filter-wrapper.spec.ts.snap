// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter-wrapper.vue should match the snapshot 1`] = `
<div class="filter__wrapper"><label class="title__field">

    <!----></label>
  <div class="filter">
    <p class="filter__label">
    <h1>Header</h1> <img src="/icons/Arrow-down.svg" alt="arrow" class="img filter__icon">
    </p>
    <p>lorem ipsum</p>
  </div>
</div>
`;
